### DISCUSSION TITLE
How to specify module/namespace so the files are generated under a directory with the module name?

### DISCUSSION THREAD

**Discussion by ingun37** [tgreyuk/typedoc-plugin-markdown#702] [Help] [answered]
I have a monorepo. I'm generating documents for one of the packages. I read if I set `--outputFileStrategy` to `members`, which is the default value, the files will be generated under a directory with the module name. So I wanted to do it but I couldn't find how. I tried adding `@modules some-name` to the top of the `index.ts`, which is my `entryPoints`, but it didn't work. It still generated everything at the top of the out directory instead of creating subdirectory with name `some-name`. Can you explain how is `module` decided so I can setup my project accordingly? Or can I set it explicitly?

**Comment by tgreyuk** [comment, marked as answer]
@ingun37 assuming i have understood correctly you can try this:

`root typedoc.json`

```
"plugin": ["typedoc-plugin-markdown"],
"packageOptions": {
    "alwaysCreateEntryPointModule": true
 }
```

and then in your index.ts (as you already mentioned):

`index.ts`
```
@module some-name
```
